### DISCUSSION TITLE
Opening notes alternatives

### DISCUSSION THREAD

**Discussion by N3C2L** [tu2-atmanand/cards-view-plugin-beta#1] [Ideas]
Thank you for the great plugin. 👍

It would be nice if you could perhaps offer other options for opening notes. If you don't mind, here are two suggestions that I would need to work better with your plugin:

1. When you click on a note in Obsidian's native file explorer, it always opens on the right side in the current tab. This means the current tab is always overwritten with the file you're clicking on when you click again. In other words, a new extra tab or tab group is never created on the right side. I would need this native file opening behavior for your plugin as well. So when I double-click with the mouse on a card in your plugin, the same opening behavior as the native file explorer should be applied. The advantage of this method is that you don't have to close the tab separately afterwards. This new option could be named something like "Open note in the right side in current tab".

2. But often I do need a new tab, so it would be advantageous for a user to have maybe a right-click menu on a card, and there is a menu whether the user wants to create the current note in a new tab in the right tab group. This menu could also be filled with a new opening option something like "Open note in the right side in a new tab". The advantage of this note opening behavior is that a user doesn't need to open many tab groups (also called splits) when double clicking the tab with the existing "Open note in the right side". A user can use the already existing split in the right side.

**Comment by tu2-atmanand** [comment]
Hii @N3C2L !

First of all thanks for using this plugin even its early stage and sharing the feedback and suggestions. The reason for not engaging into this discussion topic was because of my stupidity, the mail notification for this repo was off. And I didnt had the habit to open discussion, i directly use to jump into the issues section.

Anyways, now that I have a deep thought to the above idea, I 100% agree with you. I missed this somehow because, for exploring notes I usually use to only go through the cards and hardly use to open multiple notes at the same time. If I had to open a note to go through the content, I directly use Hover Editor plugin (my no. 1 favorite plugin). So, due to this and also, other than the development time, I havent used the opening right feature much, I didnt realized this bad experience user has to go through. The main vision I have with this plugin is itself getting hindered due to this functionality.

Now that I have came to this conclusion, I will going to make the above suggested idea as the default behavior of the plugin to open the notes on the right and use the same tab/view, instead of creating new tab. But yes, a new setting will be provided for user to make this choice.

Coming to the right-click menu suggestion, I am not sure, if ill be considering this in coming future. Mainly because I think it will make it more complicated, since there are no much things to put into this right menu yet and also right click menus are a little challenging task to achieve. Not because it difficult, integrating a stable working right click menu functionality is difficult. And I would not want to provide any thing which will raise a question on the performance and stability of this plugin. (same reason why I implemented a new way to avoid infinite scrolling, it simply didnt make any sense to me to scroll infinitely without any idea of how long a user has to scroll.)

But, if I found in furure, that after thorough testing the right click menu is working properly as expected, than it surely will be integrated.

**Comment by tu2-atmanand** [comment]
Created the following two issues after this discussion : 
- #13 
- #14 

Also, while creating above, got another idea : 
- #15

**Comment by tu2-atmanand** [comment]
Below comments are from this FR, I should have added these comments here, so sharing them again : #14 

> This feature has been partially implemented.

> Due to the limitation of APIs provided by Obsidian and also its not recommended to change the default behavior of tab management, this feature will not going to provide any special toggle button for allowing user to re-use the same tab.

> Instead, now opening the note in right side will by default going to re-use the right side tab. And if its required to open a new tab, then user has to hold CTRL key and then open the note. To make this experience even better, place the Notes explorer tab inside the left sidebar, by dragging the tab into the left side section.

> The only thing which is required to compromise here, as of now, is using the same tab for the "Open in new window" option. Based on the user requests in the future, this feature can be worked on.

**Comment by tu2-atmanand** [comment]
One important update on this FR.
The main goal of this FR was to have the functionality where we can decide whether the note should be opened in a new tab or whether it should re-use the same tab. Instead of a setting option, it will make more sense to have this functionality with some key shortcuts, since Obsidian itself, has it by default in its File Explorer. Where if you click on any file normally, it will open the note by reusing any previous tab. But if you hold down the CTRL key and then click on the filename, it will open the note in a new tab.
As per this behavior, I have tried to achieve the same in this upcoming release. But there are few limitations, as described below : 
### Layout 1 : When the Notes Explorer is opened as a normal tab
- **Open note on right side** : This option will be able to open new notes in the right split tab, by re-using the same tab if there is any previously opened, instead of creating new ones. The only limitation here is, if you have pinned the tab inside your right split and now if you try to open a new note, it will be opened inside the left split.
- **Open note in same tab** : This option cannot work in this layout. It will fallback to the first option to open the note in right split.
- **Open note in new tab** : This option will work correctly to open any note inside a new tab. But reusing the same tab is not possible in this layout, for that the second layout can be used.
- **Open in a new window** : Using this option you can open any note inside a new window, but making use of the same window to open the future notes is a little complicated task for now, hence can be worked on in the future.
### Layout 2 : When the Notes Explorer is opened inside left sidebar
- **Open note on right side** : Using this option, you can split your main workspace and the note will always be opened inside the second half. If you have pinned any note inside a second half, then it will create a new tab in that right split.
- **Open note in same tab** : Using this option you will be able to open the note by re-using the already opened tab. Just like the normal behavior of Obsidian. In this same configuration, if you want any new note should be opened inside a tab, simply hold the CTRL key and click on the card (or double click) to open it in a new tab.
- **Open note in new tab** : This will help you to open any note always inside a new tab.
- **Open in a new window** : This will work same as it works in Layout 1.

**Comment by tu2-atmanand** [comment]
I have implement this feature the best I could. Let me know if there is any improvement required and we can discuss it further here.
